pkgs/development/libraries/shibboleth-sp/default.nix

```diff
@@ -20,7 +24,7 @@ stdenv.mkDerivation rec {
     "--with-fastcgi"
   ];
 
-  env.NIX_CFLAGS_COMPILE = lib.optionalString (!stdenv.isDarwin) "-std=c++14";
+  env.NIX_CFLAGS_COMPILE = if stdenv.isDarwin then "-D_LIBCPP_ENABLE_CXX17_REMOVED_AUTO_PTR" else "-std=c++14";
```

Comment:
Looks like `"-std=c++14"` might work too? Or are there other problems?

Reply:
@wegank , I've thought so as well, but it seems not to work on darwin. More specifically, `configure` fails early with:

```
checking whether the C compiler works... no
configure: error: in '/private/tmp/nix-build-shibboleth-sp-3.0.4.1.drv-0/cpp-sp':
configure: error: C compiler cannot create executables
See 'config.log' for more details
```

Reply:
`-std=c++14` isn't a valid flag for a c compiler, thus configure fails. removing `env.NIX_CFLAGS_COMPILE` and instead adding `"CXXFLAGS=-std=c++14"` to the `configureFlags` should be sufficient.

Reply:
@annaleeleaves , thank you, works like a charm.